src/Events/Custom_Tables/V1/Migration/Events.php

```diff
@@ -83,19 +87,35 @@ public function get_ids_to_process( $limit ) {
 
 		// The lock operation could fail and that is ok. A deadlock message should not be reported in this case.
 		$suppress_errors_backup = $wpdb->suppress_errors;
-		$wpdb->suppress_errors = true;
+		$wpdb->suppress_errors  = true;
 		$wpdb->query( $lock_query );
 		$wpdb->suppress_errors = $suppress_errors_backup;
 
+		// Get our db object so we can inspect. This isn't always an object, so some type checking is needed.
+		$db = is_object( $wpdb->dbh ) ? $wpdb->dbh : null;
+
+		// Deadlock error no.
+		$deadlock_errno = 1213;
+		if ( $db !== null && $db->errno === $deadlock_errno ) {
```

Comment:
Using the `query` filter as an action, you can mock the `errno` property of the `$wpdb->dbh` object using the `uopz_set_property` function (and add its clean up to the `uopz` dedicated trait). Testing this method, then, would be fairly easy and would cover a function that would be otherwise difficult covering with tests in another way.

Reply:
Not as easy as you thought - apparently `mysqli` is somewhat protected, so you can't modify it directly. But got a workaround w/ uopz in place.